lib/beaker/host/unix/pkg.rb

```diff
@@ -17,28 +17,31 @@ def check_for_command(name)
     end
   end
 
-  def check_for_package(name)
+  def check_for_package(name, opts = {})
+    opts = {:accept_all_exit_codes => true}.merge(opts)
+    opts[:prepend_cmds] ? pc = opts.select {|k,v| k == :prepend_cmds} :
```

Comment:
After a bit more investigation it appears the reason method 'check_for_package' calls exec instead of execute is due to the need to check the exit_code value.  The execute method does however allow use of a block and we can use this to return result and still check the exit code.